Integration/AlcazarIntegration.php

```diff
@@ -188,26 +261,31 @@ public function doEnhancement(Lead $lead)
                 $params['output'] = $value;
             }
             elseif (in_array($param, ['extended', 'dnc'])) {
-                $value = $value ? 'true' : 'false';
-                $params[$param] = $value;
+                $params[$param] = $value ? 'true' : 'false';
             }
         }
        
-        $response = $this->makeRequest(
-            $keys['server'],
-            ['append_to_query' => $params],
-            'GET',
-            ['ignore_event_dispatch' => 1]
-        );       
+        try {
+            $response = $this->makeRequest(
+                $keys['server'],
+                ['append_to_query' => $params],
+                'GET',
+                ['ignore_event_dispatch' => 1]
+            );       
         
-        foreach ($response as $label => $value) {
-            $alias = 'alcazar_' . strtolower($label);
-            $default = $lead->getFieldValue($alias);
-            $lead->addUpdatedField($alias, $value, $default);        
+            foreach ($response as $label => $value) {
+                $alias = 'alcazar_' . strtolower($label);
+                $default = $lead->getFieldValue($alias);
+                $lead->addUpdatedField($alias, $value, $default);        
```

Comment:
For the record, we should probably have a whitelist of fields instead of trusting the service provider to not throw in a new one breaking our system here.